hs_core/views/__init__.py

```diff
@@ -208,6 +211,15 @@ def update_metadata_element(request, shortkey, element_name, element_id, *args,
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+@api_view(['GET'])
+def file_download_url_mapper(request, shortkey, filename):
+    """ maps the file URIs in resourcemap document to django_irods download view function"""
+
+    authorize(request, shortkey, view=True, edit=True, full=True, superuser=True)
```

Comment:
@pkdash I am thinking this authorize() call is not needed here since the same function call is done in download() view function. Also, the return from authorize() is not processed here, so I think this authorize call is not need here in this url_mapper() view. 
@pkdash @selimnairb Since Brian added api_view(['GET']) decorator in the irods download() view, I am wondering whether this decorator is needed here or not. In fact, @selimnairb if there is a way to put this api_view decorator outside django_irods download() view as Pabitra is doing here, I think it'd be better and clearer to make this decorator only added when needed as opposed to adding this decorator in the django_irods download view function,  which is shared by both api views and non-api views. I'll defer where to put api_view decorator to @pkdash and @selimnairb, but I think it'd be probably better to add this decorator only once when needed.